### PR TITLE
Add human log likelihood regularizer

### PR DESCRIPTION
**Pull request by daphne-cornelisse**
## Goal
Given a dataset of human "expert" trajectories $(o_t, a_t)$, we aim to compute the likelihood of the expert’s actions under the current policy and use this as an auxiliary learning signal.

### Dataset
Use the inverse bicycle model to infer the human actions taken for every step. 

### At initialization
The method `_save_expert_data()` saves `max_expert_samples` per env to disk. We stack actions based on `bptt_horizon` so that the resulting tuples are of shape `(max_expert_samples, bptt_horizon, action/obs_dim)`.

Under the hood - this method uses `c_collect_expert_data()` in `drive.h` to step the agents with their _continuous_ inferred human actions and collect the corresponding observations. We return the discretized actions if needed.

### Training inner loop

Do:
1. Load all data to the CPU (stored in two .pt files)
2. Sample `human_samples` tuples $(o^{\text{expert}}_t, a^{\text{expert}}_t)$ from the saved expert dataset.
3. Load this subset to the GPU 
4. Evaluate the policy likelihood $\pi^{\text{agent}}_i(a^{\text{expert}}_t \mid o^{\text{expert}}_t)$ for each transition.
5. Apply clipping and average the results.
6. Incorporate the weighted log-likelihood signal into the loss objective.

### Wandb integration

- Displays the mean log probability of human actions (negative values, closer to 0 is better)

Screenshot 2025-10-08 at 10 01 42

- Adds a new `realism/` section with the human action distributions. This is done mostly as a sanity check. The section can be extended with other realism-related metrics.
Screenshot 2025-10-08 at 11 50 18

### CODE REVIEW

File: pufferlib/config/ocean/drive.ini [hunk 1]
```diff
@@ -19,26 +19,28 @@ input_size = 256
 hidden_size = 256
 
 [env]
-num_agents = 1024
+num_agents = 32 #512
```
Comment (eugenevinitsky) [inline pufferlib/config/ocean/drive.ini:22]: Is this meant to stay here?

File: pufferlib/config/ocean/drive.ini [hunk 2]
```diff
@@ -19,26 +19,28 @@ input_size = 256
...
 total_timesteps = 2_000_000_000
 #learning_rate = 0.02
 #gamma = 0.985
 anneal_lr = True
-batch_size = 524288 # Needs to be: num_agents * num_workers * BPTT horizon
-minibatch_size = 32768
-max_minibatch_size = 32768
+batch_size = 16384
+minibatch_size = 16384
+max_minibatch_size = 16384
```
Comment (eugenevinitsky) [inline pufferlib/config/ocean/drive.ini:43]: hmm, I think I wouldn't mess with the semantics of standard parameters to get your changes in but maybe there's a nonobvious reason this is better

File: pufferlib/config/ocean/drive.ini (hunk 1 above)
Comment (daphne-cornelisse) [reply pufferlib/config/ocean/drive.ini:22]: No no it will be reverted! Currently working on this (sorry, should have marked this PR as draft!)

File: pufferlib/config/ocean/drive.ini (hunk 2 above)
Comment (daphne-cornelisse) [reply pufferlib/config/ocean/drive.ini:43]: yep 100% agree. This will all be reverted

**Comment by eugenevinitsky** [comment]
@greptile
